BinaryTree.java

```diff
@@ -153,13 +166,33 @@ public static void main(String args[])
         */
         int[] array = {2, 1, 4, 3, 5, 7};
         tree.populateTree(array);
-        tree.preorderTraversal(tree.root);
         System.out.println();
 
-        if (!tree.printAllAncestors(tree.root, 4))
+        List<TreeNode> ancestorList = new LinkedList<>();
+        if (!tree.findAllAncestors(ancestorList, tree.root, 7)) {
             System.out.println("Element doesn't exist");
+        } else {
+            System.out.println("Ancestors - " + ancestorList);
+        }
 
         System.out.println();
-        System.out.println(tree.findLowestCommonAncestor(tree.root, 4, 10));
+        System.out.print(tree.findLCAtest(7, 3));
+    }
+
+    public String findLCAtest(int a, int b) {

```

Comment:
Test methods are typically not part of the class that is being tested. It prevents polluting class' interface and lets one avoid relying on some internal data for testing.

In this case better design would be to have class BinaryTree that only implements binary tree, and another class (Main, BinaryTreeTest or similar) that would contain main() and any test code needed.